v2/internal/template_test.go

```diff
@@ -21,7 +22,25 @@ func TestParse(t *testing.T) {
 func TestParseError(t *testing.T) {
 	expectedErr := fmt.Errorf("expected error")
 	tmpl := &Template{ParseErr: &expectedErr}
-	if err := tmpl.parse("", ""); err != expectedErr {
+	if err := tmpl.parse("", "", nil); err != expectedErr {
 		t.Fatalf("expected %#v; got %#v", expectedErr, err)
 	}
 }
+
+func TestParseWithFunc(t *testing.T) {
+	tmpl := &Template{Src: "hello"}
+	funcs := template.FuncMap{
+		"foo": func() string {
```

Comment:
This test doesn't seem to actually use the "foo" func. Can you add a test that does that?